### PR TITLE
`Development`: Add continuous plagiarism control documentation

### PR DESCRIPTION
**Pull request by jakubriegel**
#### General
- [x] This is a small issue that I tested locally and was confirmed by another developer.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context
Documentation of plagiarism checks in Artemis was outdated after introduction of the continuous plagiarism control (CPC).

### Description
This PR adds a documentation of the CPC. It features general description of the feature and detailed description of flows for instructors and students.

### Steps for Testing
Read the docs and check the screenshots :) 

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### CODE REVIEW

File: docs/user/plagiarism-check.rst [hunk 1]
```diff
@@ -172,6 +173,73 @@ The process is visualized in the following diagram.
...
+
+Instructors
+^^^^^^^^^^^
+
+Configuration of the CPC
+""""""""""""""""""""""""
+
+Instructors can configure various aspects of the CPC for each exercise. The following list describes all configuration options:
+
+* **enabling the CPC** - Instructor can turn the CPC o n and off for a particular exercise. Default: `OFF`.
```
Comment (b-fein) [inline docs/user/plagiarism-check.rst:207]: Either `The instructor can […]` or `Instructors can […]` to make it a full sentence. Same for other settings below.

File: docs/user/plagiarism-check.rst [hunk 2]
```diff
@@ -172,6 +173,73 @@ The process is visualized in the following diagram.
...
+* **enabling the post due date checks** - Instructor can configure the CPC to test the exercise one more time the night after the exercise due date. This can help to test even the just in time submission. Default: `OFF`.
+* **setting the response period for the significant similarity plagiarism case** - Instructor can specify for how long after the exercise due date a student will be able to submit a complaint. Default: `7 days`. Min: `7 days`.
+* **setting the similarity threshold** - Instructor can specify the similarity threshold for the plagiarisms check performed by the CPC. Default: `90%`.
+* **setting the minimum score** - Instructor can specify the minimum submission score for the plagiarisms check performed by the CPC. Default: `0`.
+* **setting the minimum size** - Instructor can specify the minimum submission size for the plagiarisms check performed by the CPC. Default: `50`.
```
Comment (b-fein) [inline docs/user/plagiarism-check.rst:212]:
```suggestion
* **setting the similarity threshold** - Instructor can specify the similarity threshold for the plagiarism check performed by the CPC. Default: `90%`.
* **setting the minimum score** - Instructor can specify the minimum submission score for the plagiarism check performed by the CPC. Default: `0`.
* **setting the minimum size** - Instructor can specify the minimum submission size for the plagiarism check performed by the CPC. Default: `50`.
```

Comment (b-fein) [reply docs/user/plagiarism-check.rst:212]: How about rewording some of those sentences instead of essentially repeating the bold setting name in a full sentence: E.g.,

```
* **setting the minimum score** - Only submissions that have a minimum score of at least this value are included in the check for other similar submissions. Default: `0`.
* **setting the minimum size** - Submissions smaller than this value will not be checked for similarity to other submissions. Default: `50`.
```

File: docs/user/plagiarism-check.rst [hunk 3]
```diff
@@ -172,6 +173,73 @@ The process is visualized in the following diagram.
...
+
+Instructors can adjust the CPC configuration options on the exercise create and update page:
+
+|cpc-configuration|
+
+Post exercise due date steps and results verification
+"""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+After the exercise due date, instructors can manually review all of significant similarity plagiarism cases and decide on their verdicts. Steps at this point are equivalent to the steps in manual checks flow.
```
Comment (b-fein) [inline docs/user/plagiarism-check.rst:221]:
```suggestion
After the exercise due date, instructors can manually review all of the significant similarity plagiarism cases and decide on their verdicts. Steps at this point are equivalent to the steps in manual checks flow.
```

File: docs/user/plagiarism-check.rst [hunk 4]
```diff
@@ -172,6 +173,73 @@ The process is visualized in the following diagram.
...
+
+Instructors can check the plagiarism checks results CPC creates during and after the submission period using exactly the same steps and interface like with manual checks.
+
+Students
+^^^^^^^^
+
+When a student submits a work for which the CPC identifies significant similarity and creates a plagiarism case, the student receives a notification similar to what is sent when instructors create a classic plagiarism case.
+If the exercise due date has not passed, the student can submit updated work again.
+The CPC will check the submission the following night and will either remove or update the significant similarity plagiarism case based on the results of the check.
+Before the exercise due date student cannot see submission of other students involved in the significant similarity plagiarism case.
```
Comment (b-fein) [inline docs/user/plagiarism-check.rst:231]:
```suggestion
Before the exercise due date, student cannot see the submissions of other students involved in the significant similarity plagiarism case.
```

File: docs/user/plagiarism-check.rst [hunk 5]
```diff
@@ -42,7 +43,7 @@ Before starting the plagiarism check, the user can configure different settings
 
 3. Minimum Size.
 
-  a. Default value: Consider only submissions whose size is greater or equal to this value.
+  a. Programing exercises: Consider only submissions that have at least as many `git diff` lines in comparison to the template as the specified value.
```
Comment (dearjasmina) [inline docs/user/plagiarism-check.rst:46]: "Programming" instead of Programing (just a typo)

File: docs/user/plagiarism-check.rst [hunk 6]
```diff
@@ -69,18 +70,18 @@ Results
 ^^^^^^^
 After the plagiarism check was executed, the results can be inspected in different views.
 
-1. Overview of the similarity distribution. This statistical overview shows the similarity distribution based on the percentage as a histogram chart. The user can analyze the distribution quickly and adjust the plagiarism check settings as needed accordingly.
+1. Overview of the similarity distribution. This statistical overview shows the similarity distribution based on the percentage as a histogram chart. Above the chart there are additional metrics showing various statistics about the latest check. The user can analyze the distribution quickly and adjust the plagiarism check settings as needed accordingly.
```
Comment (dearjasmina) [inline docs/user/plagiarism-check.rst:73]: I would rephrase as: "Above the chart, additional metrics show various statistics about the latest check. The user can analyze the distribution quickly and adjust the plagiarism check settings accordingly."

File: docs/user/plagiarism-check.rst [hunk 7]
```diff
@@ -172,6 +173,73 @@ The process is visualized in the following diagram.
 2. Reply to the message of the instructor.
 3. Wait for the final verdict.
 
+Continuous Plagiarism Control
+-----------------------------------------
+
+In this section, we explain the automated flow of the continuous plagiarism control (CPC) and presents actions taken by from different Artemis users.
```
Comment (dearjasmina) [inline docs/user/plagiarism-check.rst:179]: - "present" instead of presents
- you can delete "from"

File: docs/user/plagiarism-check.rst [hunk 8]
```diff
@@ -172,6 +173,73 @@ The process is visualized in the following diagram.
...
+Instructors
+^^^^^^^^^^^
+
+Configuration of the CPC
+""""""""""""""""""""""""
+
+Instructors can configure various aspects of the CPC for each exercise. The following list describes all configuration options:
+
+* **enabling the CPC** - Instructor can turn the CPC o n and off for a particular exercise. Default: `OFF`.
+* **enabling the post due date checks** - Instructor can configure the CPC to test the exercise one more time the night after the exercise due date. This can help to test even the just in time submission. Default: `OFF`.
```
[the remark this answers is not in the document]
Comment (dearjasmina) [reply docs/user/plagiarism-check.rst:208]: I'm also not sure what's meant here, if I understood it correctly I would rephrase it as:
The instructor can configure the CPC to test the exercise one more time the night after the exercise due date. This can test even the "just-in-time" submissions. Default: `OFF`.

[the remark this answers is not in the document]
Comment (jakubriegel) [reply docs/user/plagiarism-check.rst:208]: Changed to: `This allows to test submission posted shortly before the due date.`

File: docs/user/plagiarism-check.rst (hunk 2 above)
Comment (jakubriegel) [reply docs/user/plagiarism-check.rst:212]: Those three options are described above in the manual checks section. Here I want to explicitly mention their name, so that it is clear that these are the same

File: docs/user/plagiarism-check.rst [hunk 9]
```diff
@@ -172,6 +173,73 @@ The process is visualized in the following diagram.
...
+^^^^^^^^^^^^^^
+
+The CPC is an experimental feature designed to shorten the feedback loop for students and reduce the manual work required from tutors and instructors.
+Continuous feedback on potential plagiarism aims to encourage students to submit their original work and enhance their learning outcomes.
+The continuous plagiarism control **does not accuse students of plagiarism, nor does it affect their scores**.
+Instead, it **notifies students of significant similarities** detected between their submissions, allowing them to revise their solutions.
+
+The CPC initiates plagiarism checks every night for all onboarded exercises, processing all latest submissions for each exercise.
+If a submission does not include significant similarity, the CPC removes any associated existing plagiarism case.
+If significant similarity is detected, the CPC creates or updates a plagiarism case for that submission.
```
[the remark this answers is not in the document]
Comment (jakubriegel) [reply docs/user/plagiarism-check.rst:191]: I've changed the description to explicitly mention `significant similarity plagiarism case`. This should be more clear for the user

**Review by LenaKahle — DISMISSED** [review]
Approve, looks good.

**Review by xHadie — APPROVED** [review]
Very clear and easily understandable 👍🏼

File: docs/user/plagiarism-check.rst [hunk 10]
```diff
@@ -172,6 +173,73 @@ The process is visualized in the following diagram.
...
+Automated Flow
+^^^^^^^^^^^^^^
+
+The CPC is an experimental feature designed to shorten the feedback loop for students and reduce the manual work required from tutors and instructors.
+Continuous feedback on potential plagiarism aims to encourage students to submit their original work and enhance their learning outcomes.
+The continuous plagiarism control **does not accuse students of plagiarism, nor does it affect their scores**.
+Instead, it **notifies students of significant similarities** detected between their submissions, allowing them to revise their solutions.
+
+The CPC initiates plagiarism checks every night for all onboarded exercises, processing all the latest submissions for each exercise.
+If a significant similarity is detected, the CPC creates or updates a significant similarity plagiarism case for that submission.
```
Comment (dfuchss) [inline docs/user/plagiarism-check.rst:190]: What is a `significant similarity plagiarism case` and how does it relate to "normal" plagiarism cases?

File: docs/user/plagiarism-check.rst [hunk 11]
```diff
@@ -172,6 +173,73 @@ The process is visualized in the following diagram.
...
+^^^^^^^^^^^
+
+Configuration of the CPC
+""""""""""""""""""""""""
+
+Instructors can configure various aspects of the CPC for each exercise. The following list describes all configuration options:
+
+* **Instructors can enable the CPC.** - Instructor can turn the CPC on and off for a particular exercise. Default: `OFF`.
+* **Instructors can enable the post due date checks.** - Instructor can configure the CPC to test the exercise one more time the night after the exercise due date. This allows to test submission posted shortly before the due date. Default: `OFF`.
+* **Instructors can set the response period for the significant similarity plagiarism case.** - Instructor can specify for how long after the exercise due date a student will be able to submit a complaint. Default: `7 days`. Min: `7 days`.
```
Comment (dfuchss) [inline docs/user/plagiarism-check.rst:209]: Why do students have to do complaints? In the beginning, you state "The continuous plagiarism control **does not accuse students of plagiarism, nor does it affect their scores**"

File: docs/user/plagiarism-check.rst [hunk 12]
```diff
@@ -172,6 +173,73 @@ The process is visualized in the following diagram.
...
+* **Instructors can set the similarity threshold.** - Instructor can specify the similarity threshold for the plagiarism check performed by the CPC. Default: `90%`.
```
Comment (dfuchss) [inline docs/user/plagiarism-check.rst:210]: Please add, that this value is highly depended on the actual task. Also add that similarities greater than that do not necessarily mean that they are plagiarism

File: docs/user/plagiarism-check.rst [hunk 13]
```diff
@@ -172,6 +173,73 @@ The process is visualized in the following diagram.
...
+* **Instructors can set the minimum score.** - Instructor can specify the minimum submission score for the plagiarism check performed by the CPC. Default: `0`.
+* **Instructors can set the minimum size.** - Instructor can specify the minimum submission size for the plagiarism check performed by the CPC. Default: `50`.
```
Comment (dfuchss) [inline docs/user/plagiarism-check.rst:212]: What is the size? LoC ?

File: docs/user/plagiarism-check.rst [hunk 14]
```diff
@@ -172,6 +173,73 @@ The process is visualized in the following diagram.
...
+
+Instructors can adjust the CPC configuration options on the exercise create and update page:
+
+|cpc-configuration|
+
+Post exercise due date steps and results verification
+"""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+After the exercise due date, instructors can manually review all of the significant similarity plagiarism cases and decide on their verdicts. Steps at this point are equivalent to the steps in manual checks flow.
```
Comment (dfuchss) [inline docs/user/plagiarism-check.rst:221]: Just to be sure: What happens if they don't do anything? Noting ?

File: docs/user/plagiarism-check.rst [hunk 15]
```diff
@@ -172,6 +173,73 @@ The process is visualized in the following diagram.
...
+
+Instructors can check the plagiarism checks results CPC creates during and after the submission period using exactly the same steps and interface like with manual checks.
+
+Students
+^^^^^^^^
+
+When a student submits a work for which the CPC identifies significant similarity and creates a plagiarism case, the student receives a notification similar to what is sent when instructors create a classic plagiarism case.
```
Comment (dfuchss) [inline docs/user/plagiarism-check.rst:228]: I think you should separate the two concepts: Plagiarism Case & Significant Similarity. See comment above :)

File: docs/user/plagiarism-check.rst [hunk 16]
```diff
@@ -172,6 +173,73 @@ The process is visualized in the following diagram.
...
+If the exercise due date has not passed, the student can submit updated work again.
+The CPC will check the submission the following night and will either remove or update the significant similarity plagiarism case based on the results of the check.
```
Comment (dfuchss) [inline docs/user/plagiarism-check.rst:230]: Does that mean, that the message will be shown to the student all the time?

File: docs/user/plagiarism-check.rst [hunk 17]
```diff
@@ -172,6 +173,73 @@ The process is visualized in the following diagram.
...
+Before the exercise due date, a student cannot see the submissions of other students involved in the significant similarity plagiarism case.
```
Comment (dfuchss) [inline docs/user/plagiarism-check.rst:231]: Just for clarification: Afterwards they can?

File: docs/user/plagiarism-check.rst [hunk 18]
```diff
@@ -172,6 +173,73 @@ The process is visualized in the following diagram.
...
+
+When a significant similarity plagiarism case exists for the latest submission, the student sees a dedicated button redirecting to it:
+|significant-similarity-button|
+
+A significant similarity plagiarism case contains a short message and an overview of similarity in the submission:
+|significant-similarity-plagiarism-case|
+
+In case the exercise due date has passed and student does not agree with the significant similarity plagiarism case,
```
Comment (dfuchss) [inline docs/user/plagiarism-check.rst:239]: So do you deduct points after the due date?

**Review by dfuchss — COMMENTED** [review]
Just some questions and comments for clarification :)

**Review by MaximilianJG — DISMISSED** [review]
Very clear. Very understandable. The diagrams help convey the topic well. Good job!

File: docs/user/plagiarism-check.rst (hunk 10 above)
Comment (jakubriegel) [reply docs/user/plagiarism-check.rst:190]: I added a sentence explaining what it is

File: docs/user/plagiarism-check.rst (hunk 11 above)
Comment (jakubriegel) [reply docs/user/plagiarism-check.rst:209]: In the initial description of the CPC flow I added a short explanation that eventually instructors make decision about accusing student for plagiarism or not

File: docs/user/plagiarism-check.rst (hunk 13 above)
Comment (jakubriegel) [reply docs/user/plagiarism-check.rst:212]: Depends on the exercise type. For programming exercises it is the number of diff lines between the submission and the template. It is documented in one of previous sections and in the pop ups in the client

File: docs/user/plagiarism-check.rst (hunk 15 above)
Comment (jakubriegel) [reply docs/user/plagiarism-check.rst:228]: Explained above

File: docs/user/plagiarism-check.rst (hunk 17 above)
Comment (jakubriegel) [reply docs/user/plagiarism-check.rst:231]: Yes

File: docs/user/plagiarism-check.rst (hunk 18 above)
Comment (jakubriegel) [reply docs/user/plagiarism-check.rst:239]: Nothing happens automatically. Instructors make decisions about any steps after the deadline

**Review by xHadie — DISMISSED** [review]

**Review by LenaKahle — APPROVED** [review]
Reapprove.

File: docs/user/plagiarism-check.rst [hunk 19]
```diff
@@ -69,18 +70,18 @@ Results
 ^^^^^^^
 After the plagiarism check was executed, the results can be inspected in different views.
 
-1. Overview of the similarity distribution. This statistical overview shows the similarity distribution based on the percentage as a histogram chart. The user can analyze the distribution quickly and adjust the plagiarism check settings as needed accordingly.
+1. Overview of the similarity distribution. This statistical overview shows the similarity distribution based on the percentage as a histogram chart. Above the chart, additional metrics show various statistics about the latest check. The user can analyze the distribution quickly and adjust the plagiarism check settings accordingly.
 
  |run-results|
 
 2. Selected submission. When the user selects the submission from the submission list on the left side of the plagiarism run result view, new actions can be selected to initiate the plagiarism check workflow. We will provide further details to the workflow in the next section.
```
Comment (Ceiridge) [inline docs/user/plagiarism-check.rst:77]: details to the workflow -> on the workflow

File: docs/user/plagiarism-check.rst [hunk 20]
```diff
@@ -172,6 +173,75 @@ The process is visualized in the following diagram.
...
+The continuous plagiarism control **does not accuse students of plagiarism, nor does it affect their scores**.
+Instead, it **notifies students of significant similarities** detected between their submissions, allowing them to revise their solutions.
+
+The CPC initiates plagiarism checks every night for all onboarded exercises, processing all the latest submissions for each exercise.
+If a significant similarity is detected, the CPC creates or updates a significant similarity plagiarism case for that submission.
+It is a special form of plagiarism case, which is marked as a warning and presents a gentle message about detected similarity.
+If a submission does not include significant similarity, the CPC removes any associated existing significant similarity plagiarism case.
+Subsequently, the student analyzes the feedback received and, if the submission period is still open, can submit a new solution.
+If the submission period has ended, the student can either dispute the plagiarism case or choose to take no action.
+Eventually, the instructor makes a decision about accusing student of plagiarism and about any further steps.
```
Comment (Ceiridge) [inline docs/user/plagiarism-check.rst:195]: accusing **the** student

File: docs/user/plagiarism-check.rst [hunk 21]
```diff
@@ -172,6 +173,75 @@ The process is visualized in the following diagram.
...
+Instructors
+^^^^^^^^^^^
+
+Configuration of the CPC
+""""""""""""""""""""""""
+
+Instructors can configure various aspects of the CPC for each exercise. The following list describes all configuration options:
+
+* **Instructors can enable the CPC.** - Instructor can turn the CPC on and off for a particular exercise. Default: `OFF`.
+* **Instructors can enable the post due date checks.** - Instructor can configure the CPC to test the exercise one more time the night after the exercise due date. This allows to test submission posted shortly before the due date. Default: `OFF`.
```
Comment (Ceiridge) [inline docs/user/plagiarism-check.rst:210]: post due -> post-due

Comment (Ceiridge) [inline docs/user/plagiarism-check.rst:210]: The last sentence is very hard to understand. Does this refer to multiple submissions or is it passive tone for one submission (be posted)?

**Review by Ceiridge — CHANGES_REQUESTED** [review]
Great work, but there are some minor grammar mistakes found with the help of Grammarly

File: docs/user/plagiarism-check.rst (hunk 21 above)
Comment (jakubriegel) [reply docs/user/plagiarism-check.rst:210]: It's describes a general case where the submissions are made close to the dead line. Changed to `This allows to test **submissions** posted shortly before the due date`.

**Review by Ceiridge — APPROVED** [review]

**Review by LenaKahle — APPROVED** [review]
Reapprove.

**Review by xHadie — APPROVED** [review]

File: docs/user/plagiarism-check.rst (hunk 16 above)
Comment (dfuchss) [reply docs/user/plagiarism-check.rst:232]: I mean, does it disappear after you submit another version?

**Review by dearjasmina — APPROVED** [review]
